src/main/java/com/dev/cinema/dao/impl/ShoppingCartDaoImpl.java

```diff
@@ -49,7 +49,8 @@ public ShoppingCart getByUser(User user) {
         try (Session session = sessionFactory.openSession()) {
             Query<ShoppingCart> shoppingCartQuery = session
                     .createQuery("from ShoppingCart sc "
-                            + "join fetch sc.user "
+                            + "join fetch sc.user u "
+                            + "join fetch u.roles "
```

Comment:
I think it is redundant to fetch roles here, we do it when login in, and then we operate with the UserDetail object and use roles from that object. For adding sessions you don't need to get roles from DB. But you can ask on QnA ^_^